packages/health/example/lib/util.dart

```diff
@@ -68,12 +68,12 @@ const List<HealthDataType> dataTypesAndroid = [
   // HealthDataType.MOVE_MINUTES, // TODO: Find alternative for Health Connect
   HealthDataType.DISTANCE_DELTA,
   HealthDataType.RESPIRATORY_RATE,
-  HealthDataType.SLEEP_AWAKE,
-  HealthDataType.SLEEP_ASLEEP,
-  HealthDataType.SLEEP_LIGHT,
-  HealthDataType.SLEEP_DEEP,
-  HealthDataType.SLEEP_REM,
-  HealthDataType.SLEEP_SESSION,
+  // HealthDataType.SLEEP_AWAKE,
```

Comment:
Are these changes relevant?

Reply:
Fetching the SLEEP data is crashing the app with the same error as writing.

Reply:
I checked out this PR, uncommented the SLEEP lines, and i can read sleep data without a crash.

Reply:
@eliasteeny could you please double check if this error still occurs on your device and if so, tell us your OS version and setup?

Reply:
I got a closer look. I did not have a wearable so was reading only sleep session data (time asleep, one data point) and that worked fine. I get the crash now that i have a wearable and i try to read the stages. The issue seems to be that SleepStageRecord object no longer exists in Android 14. Instead there is a list in the SleepSessionRecord. You can see this sample code in https://developer.android.com/health-and-fitness/guides/health-connect/develop/sessions

suspend fun readSleepSessions(
    healthConnectClient: HealthConnectClient,
    startTime: Instant,
    endTime: Instant
) {
    val response =
        healthConnectClient.readRecords(
            ReadRecordsRequest(
                SleepSessionRecord::class,
                timeRangeFilter = TimeRangeFilter.between(startTime, endTime)
            )
        )
    for (sleepRecord in response.records) {
        // Retrieve relevant sleep stages from each sleep record
        val sleepStages = sleepRecord.stages
    }
}

The convertRecord function returns a single record and now it should return many, not sure how to handle this with the way it works now?